flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -2729,11 +2727,11 @@ object JobManager {
   def getRemoteJobManagerAkkaURL(config: Configuration) : String = {
     val (protocol, hostname, port) = TaskManager.getAndCheckJobManagerAddress(config)
 
-    var hostPort: InetSocketAddress = null
+    var hostPort: InetSocketAddress = new InetSocketAddress(hostname, port)
```

Comment:
It is just a check to see if we expect messages from a resolvable hostname. It is not necessary to do that but I found test cases rely on that. Also, it improves error reporting for the user in case a non-resolvable hostname was chosen.